Sources/GRPC/GRPCServerRequestRoutingHandler.swift

```diff
@@ -57,7 +56,7 @@ public struct CallHandlerContext {
 /// from the pipeline.
 public final class GRPCServerRequestRoutingHandler {
   private let logger: Logger
-  private let servicesByName: [String: CallHandlerProvider]
+  private let servicesByName: [Substring: CallHandlerProvider]
```

Comment:
Not directly related to these changes: but I wonder if just storing `[ChannelHandlerProvider]` would be more efficient here. The provider stores enough information and I'd expect the number of providers to be low.

Reply:
Yeah, in principle we could investigate that as a further optimisation as well.